### ISSUE TITLE
New to DOSBox-X , coming from DOS, DOSBox, and Boxer

### ISSUE THREAD

**Issue by blogabe** [joncampbell123/dosbox-x#2062]
**Please describe your question.**
A clear and concise description of what your question is. Ex. How do I [...]
Hi! I'm new to DOSBox-X and am annoyed I hadn't come across it previously. It's a much better experience for me and what I want to do. I do have a few questions I'm hoping you can help me out with. BTW: I'm using macOS Catalina.

1. The DOSBox-X preferences file is not saved anywhere on my machine including **~/Library/Preferences/**. I've also checked root, home, and /Applications (which is where DOSBox-X exists on my system). I ended up copying the pref file from the code and wonder if this is the preferred way or if it's user error on my part?
2. Is there an equivalent of DOSKEY to set aliases? I see a couple open issues here and the command appears to not be available. I'm more interested in setting an alias and not particularly tied to DOSKEY.
3. I'm trying to install dBase 3p which is a multi-disk install. Unlike other multi disk installs, it's installer doesn't seem to pick up the floppy disk image remount to the next disk and I'm continually greeted w/ a "insert disk 2 and press any key" message. This only happens on dBase 3 from what I've used so far.
4. When the above happens, I'm stuck. Sending ctrl-break and ctrl-esc do not exit me out of the installer, but ctrl-alt-del does reboot. Anything to be aware of?

Appreciate your help. Thank you.

**Comment by joncampbell123** [comment]
1. There has been some contributions that affect where it is saved, but at this moment, I'm not sure. @Wengier @emendelson what do recent changes do here?
2. There is a "swap disk" mapper shortcut. If you imgmount to a floppy drive and list all disk images, you should be able to swap between as needed.
3. If the installer has hooked CTRL+Break or CTRL+C, then those keys should do what the installer intends them to do. In this case, it sounds like it's designed to ignore them. CTRL+Esc is not a key combination handled by DOSBox-X unless you assign something to it in the mapper.

**Comment by joncampbell123** [comment]
Also from my email, DOSBox-X, DOSBox SVN, etc. emulate DOSKEY-like command line input but the full DOSKEY API is not implemented.

**Comment by Wengier** [comment]
In the latest DOSBox-X 0.83.8 when you try to save the config file from the built-in configuration tool, you will see the following options:

Use primary config file
Use portable config file
Use user config file

The primary config file refers to the config file currently being used (if any), and the portable config file refers to dosbox-x.conf in the current directory. On the other hand, if you click "Use user config file", then the config file will become ~/Library/Preferences/DOSBox-X 0.83.8 Preferences

You can see the current config file being used by entering the command "config -l" from the DOSBox-X's DOS shell.

Also, as @joncampbell123 already mentioned, if you want to do a multi-disk install, you don't need to unmount and then remount floppy disk images. Just mount all disk images at once to Drive A:, and then swap the disk from the menu. This will make it much easier for the multi-disk install.

Moreover, there is an ALIAS command in DOSBox-X which allows you to do aliasing in the command shell.

**Comment by blogabe** [comment]
Awesome. Thank you both very much. I'll spend some more time learning the ins and outs.

Separately I see WPDos is now using Dosbox-x under the hood so i may end up using that "drive" to add other software.

**Comment by blogabe** [comment]
I ended up using `alias` which is exactly what i was looking for. I had tried it but received an error message initially so assume user error the first time.

I was using brews version which was .7 but I see they upgraded to .8 now. Will try it out. Thanks again!

**Comment by blogabe** [comment]
The issue I'm seeing w/ multi image mount is that it won't mount, I receive a `Drive A: failed to mount.` error.

**Comment by Wengier** [comment]
@blogabe Can you try mount multiple image files from the shell and see the exact issue, e.g.

IMGMOUNT disk1.img disk2.img disk3.img

**Comment by blogabe** [comment]
That seems to work. The image file has spaces in it. Escaping the space in the shell didn't work, but wrapping them in double quotes did. Working through rest of installation now

**Comment by Wengier** [comment]
@blogabe I tried to mount two simple image files with spaces (named "disk test 1.img" and "disk test 2.img") from the menu and it seems to work. Maybe the names of your image files are more complicated. Perhaps you can post the names and let me see why they may fail?

**Comment by blogabe** [comment]
Yeah I ended up playing around with this last night. I don't believe the spaces were the problem as I experienced the same issue pointing to multiple img files from the menubar. They all worked fine when mounting from the shell.

Saying that, I was having issues w/ installing Turbo Assembler (too many subdirectories error) and dBase3+ (install couldn't deal with the disk swapping) with these methods. I ended up installing and copying them from my virtualized environment in Fusion to a floppy img file and then mounted that img file then copying it over to DosBox-X's c-drive.

Only remaining hurdle is to get Lotus 123 working. Installs fine, but I believe the WYSIWIG is making it so that DosBox-X won't full load beyond the startup screen.

**Comment by emendelson** [comment]
> 1. The DOSBox-X preferences file is not saved anywhere on my machine including **~/Library/Preferences/**. I've also checked root, home, and /Applications (which is where DOSBox-X exists on my system). I ended up copying the pref file from the code and wonder if this is the preferred way or if it's user error on my part?

DOSBox-X doesn't create a macOS-style preference file anywhere. If you enter this command at the prompt in DOSBox-X -

`config -wc`

it will write a `dobox-x.conf` file to your home folder. If you edit the file, DOSBox-X will load the edited version.

**Comment by blogabe** [comment]
Thank you @emendelson. Appreciate the help. I ended up using the Config Tool to create the user config file under ~/Library/Preferences although I'm thinking about switching to the portable config file in my home directory. Very much enjoying the DosBox-X experience so far. Had some time to mess around w/ it yesterday on my wife's new Silicon Air and so far so good.
